packages/openapi-generator/src/file-serializer/api-file.spec.ts

```diff
@@ -122,14 +133,16 @@ describe('apiFile', () => {
          * Create a request builder for execution of get requests to the 'test/{id}' endpoint.
          * @param id - Path parameter.
          * @param queryParameters - Object containing the following keys: queryParam.
+         * @param headerParameters - Object containing the following keys: headerParam.
          * @returns The request builder, use the \`execute()\` method to trigger the request.
          */
-        getFn: (id: string, queryParameters: {'queryParam': QueryParameterType}) => new OpenApiRequestBuilder<string>(
+        getFn: (id: string, headerParameters?: {'headerParam'?: string}, queryParameters: {'queryParam': QueryParameterType}) => new OpenApiRequestBuilder<string>(
```

Comment:
[req] This seems problematic to me. The optional header parameters cannot come before the required query paramters. This is not allowed. I wonder if this is how our API is intended? I would also assume headers come last, however, it is possible that headers will be required and query parameters are optional. I somehow thought this was all in one object.

Reply:
Should be good now. Also added a case where the optional modifier will be added to queryParameters object if there are no headerParameters defined as this was not covered.